cpmpy/transformations/normalize.py

```diff
@@ -34,3 +37,94 @@ def unravel(lst, append):
     unravel((cpm_expr,), append)
 
     return newlist
+
+
+def simplify_boolean(lst_of_expr, num_context=False):
+    """
+    removes boolean constants from all CPMpy expressions
+    only resulting boolean constant is literal 'false'
+    - list_of_expr: list of CPMpy expressions
+    """
+
+    newlist = []
+    for expr in lst_of_expr:
+
+        if isinstance(expr, bool):
+            # not sure if this should happen here or at construction time
+            expr = BoolVal(expr)
+
+        if isinstance(expr, BoolVal):
+            newlist.append(int(expr.value()) if num_context else expr)
+
+        elif isinstance(expr, Operator):
+            args = simplify_boolean(expr.args, num_context=not expr.is_bool())
+
+            if expr.name == "or":
+                if any(is_false_cst(arg) for arg in args):
+                    newlist.append(1 if num_context else BoolVal(True))
+                else:
+                    newlist.append(Operator("or", [arg for arg in args if not isinstance(arg, BoolVal)]))
```

Comment:
I think this would not clean a 'true' or np.true; I also think we are needlessly copying a lot of objects that do not change.

But I also don't have a solution straight away... will think about it

Reply:
I could just check for `is_false_cst` instead of the `BoolVal` here? (The `is_false_cst` on line 63 should be `is_true_cst`!) That check will also hold for np constants.

Not sure if we can do anything about the copying... Except for changing args in place but some transformation will have to copy them at least once as the users constraints should remain unchanged